acceptFactory.cc

```diff
@@ -102,10 +178,162 @@ void acceptItem::remakeConnection()
+        if(sep2==spec.npos || sep3==spec.npos) {
+            fprintf(stderr, "Unix path+permissions spec unparsable '%s'\n", path);
+            exit(1);
+        }
+        std::string user(spec.substr(0,sep)),
+                    grp (spec.substr(sep+1, sep2-sep-1)),
+                    perm(spec.substr(sep2+1, sep3-sep2-1));
+
+        if(!user.empty()) {
+            struct passwd *uinfo = getpwnam(user.c_str());
+            if(!uinfo) {
+                fprintf(stderr, "Unknown user '%s'\n", user.c_str());
+                exit(1);
+            }
+            uid = uinfo->pw_uid;
+            gid = uinfo->pw_gid;
+        }
+
+        if(!grp.empty()) {
+            struct group *ginfo = getgrnam(grp.c_str());
+            if(!ginfo) {
+                fprintf(stderr, "Unknown group '%s'\n", grp.c_str());
+                exit(1);
+            }
+            gid =ginfo->gr_gid;
+        }
+
+        if(!perm.empty()) {
+            perms = strtoul(perm.c_str(), NULL, 8);
+        }
+
+        spec = spec.substr(sep3+1);
+    }
+
+    if(spec.empty()) {
+        fprintf(stderr, "expected unix socket path spec.\n");
+        exit(1);
+    }
+
+    /* Abstract unix sockets are a linux specific feature whereby
+     * the socket has a "path" name associated with it, but no presence
+     * in the filesystem (and no associated permission check).
+     * Analogous to IP bound to localhost with a name string
+     * instead of a port number.
+     * We denote an abstract socket with a leading '@'.
+     */
+    abstract = spec[0]=='@';
+#ifndef __linux__
+    if(abstract) {
+        fprintf(stderr, "Abstract unix sockets not supported by this host\n");
+        exit(1);
+    }
+#endif
+
+    memset(&addr, 0, sizeof(0));
+    addr.sun_family = AF_UNIX;
+    if(spec.size()>=sizeof(addr.sun_path)) {
+        fprintf(stderr, "Unix path is too long (must be <%zu)\n", sizeof(addr.sun_path));
+        exit(1);
+    }
+
+    memcpy(addr.sun_path, spec.c_str(), spec.size()+1);
+
+    PRINTF("Created new telnet UNIX listener (acceptItem %p) at '%s' (read%s)\n",
+           this, addr.sun_path, readonly?"only":"/write");
+
+    /* signal an abstract socket with a *leading* nil.
+     * We replace the '@'
+     */
+    if(abstract)
+        addr.sun_path[0] = '\0';
+
+    remakeConnection();
+}
+
+
+acceptItemUNIX::~acceptItemUNIX()
+{
+    if(!abstract)
+        unlink(addr.sun_path);
+}
+
+void acceptItemUNIX::remakeConnection()
+{
+    int bindStatus;
+
+    if(!abstract && unlink(addr.sun_path) < 0) {
+        if(errno!=ENOENT) {
+            PRINTF("Failed to remove unix socket at '%s' : %s\n", addr.sun_path, strerror(errno));
+            throw errno; // Nooooo!
+        }
+    }
+
+    _fd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (_fd < 0) {
+        PRINTF("Socket error: %s\n", strerror(errno));
+        throw errno;
+    }
+
+    bindStatus = bind(_fd, (struct sockaddr *) &addr, sizeof(addr));
```

Comment:
ha, right.  From the manpage

> Null  bytes in the name have no special significance.

Looks like in both cases the length could be set by strlen().  I'll try this.